File: app/src/main/java/com/jormy/nin/Reflector.java
package com.jormy.nin;

import java.lang.reflect.Method;

/* loaded from: classes.dex */
class Reflector {
    Reflector() {
    }

    public static String analyzeClass(String leclassname) {
        try {
            Class leclass = Class.forName(leclassname);
            StringBuilder bi = new StringBuilder();
            Method[] aClassMethods = leclass.getDeclaredMethods();
            for (Method m : aClassMethods) {
                bi.append(m.toGenericString());
                bi.append("\n");
            }
            return bi.toString();
        } catch (ClassNotFoundException e) {
            return "-- class not found!";
        } catch (Exception e2) {
            return "-- exception : " + e2.toString();
        }
    }
}

File: app/src/main/java/com/jormy/nin/Reflector.kt
package com.jormy.nin

@Api
internal object Reflector {

    @Api
    @JvmStatic
    fun analyzeClass(className: String): String {
        return try {
            val theClass = Class.forName(className)
            val builder = StringBuilder()
            for (method in theClass.declaredMethods) {
                builder.append(method.toGenericString())
                builder.append("\n")
            }
            builder.toString()
        } catch (e: ClassNotFoundException) {
            "-- class not found!"
        } catch (e2: Exception) {
            "-- exception : $e2"
        }
    }
}
